### PR TITLE
Remove FKs blocking minimal bootstrap config

### PR DESCRIPTION
**Pull request by connorwstein**
Suspect these FK constraints were causing problems as soon as we started doing spec validation in https://github.com/smartcontractkit/chainlink/pull/3597/files#diff-c0433b0f326c510169216d4fbf3db8daa1024b4901449f1f38b4a7bba6b42052R590. Removed them so bootstrap nodes can be specified without.

### CODE REVIEW

File: core/services/validators.go [hunk 1]
```diff
@@ -394,6 +396,7 @@ func ValidatedOracleSpec(tomlString string) (offchainreporting.OracleSpec, error
 			ContractConfigTrackerPollInterval:      models.Interval(1 * time.Minute),
 			ContractConfigConfirmations:            uint16(3), // TODO: why a uint16? just forcing casting everywhere
 		},
+		Pipeline: *pipeline.NewTaskDAG(),
```
Comment (connorwstein) [inline core/services/validators.go:399]: Fix for the issue around not specifying an observationSource in a bootstrap node causing a panic

File: core/services/offchainreporting/oracle.go [hunk 2]
```diff
@@ -182,6 +177,11 @@ func (d jobSpawnerDelegate) ServicesForSpec(spec job.Spec) ([]job.Service, error
 		}
 
 	} else {
+		ocrkey, exists := d.keyStore.DecryptedOCRKey(concreteSpec.EncryptedOCRKeyBundleID)
```
Comment (connorwstein) [inline core/services/offchainreporting/oracle.go:180]: OCR key bundles are only relevant for non-bootstrap nodes.

File: core/store/migrations/migration1605218542/migrate.go [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+import "github.com/jinzhu/gorm"
+
+func Migrate(tx *gorm.DB) error {
+	return tx.Exec(`
+        ALTER TABLE offchainreporting_oracle_specs DROP CONSTRAINT offchainreporting_oracle_specs_encrypted_ocr_key_bundle_id_fkey;
+        ALTER TABLE offchainreporting_oracle_specs DROP CONSTRAINT offchainreporting_oracle_specs_transmitter_address_fkey;
+        ALTER TABLE offchainreporting_oracle_specs ALTER COLUMN transmitter_address DROP NOT NULL;
+        ALTER TABLE offchainreporting_oracle_specs ALTER COLUMN encrypted_ocr_key_bundle_id DROP NOT NULL;
+        ALTER TABLE offchainreporting_oracle_specs ADD CONSTRAINT encrypted_ocr_key_bundle_id_not_null CHECK((is_bootstrap_peer) OR (encrypted_ocr_key_bundle_id IS NOT NULL));
```
Comment (connorwstein) [inline core/store/migrations/migration1605218542/migrate.go:11]: OCR key bundles are only relevant for non-bootstrap nodes.

File: core/services/offchainreporting/oracle.go (hunk 2 above)
Comment (connorwstein) [reply core/services/offchainreporting/oracle.go:180]: also importantly, this is the only place EncryptedOCRKeyBundleID is used

**Review by spooktheducks — APPROVED** [review]

**Pull request merged**